### ISSUE TITLE
Generate images but got images with noises

### ISSUE THREAD

**Issue by gyNancy** [lucidrains/imagen-pytorch#187]
Hi,

When I try to generate images using the given first example code, I saved the images but got images with random noises (I trained imagen with 1000 times). I'm new to DALLE, and is this how it supposed to work? I thought I should have somewhat images although the quality might not be perfect. Any help would be good. Thanks. 
import torch
from imagen_pytorch import Unet, Imagen

# unet for imagen

unet1 = Unet(
    dim = 32,
    cond_dim = 512,
    dim_mults = (1, 2, 4, 8),
    num_resnet_blocks = 3,
    layer_attns = (False, True, True, True),
    layer_cross_attns = (False, True, True, True)
)

unet2 = Unet(
    dim = 32,
    cond_dim = 512,
    dim_mults = (1, 2, 4, 8),
    num_resnet_blocks = (2, 4, 8, 8),
    layer_attns = (False, False, False, True),
    layer_cross_attns = (False, False, False, True)
)

# imagen, which contains the unets above (base unet and super resoluting ones)

imagen = Imagen(
    unets = (unet1, unet2),
    image_sizes = (64, 256),
    timesteps = 1000,
    cond_drop_prob = 0.1
).cuda()

# mock images (get a lot of this) and text encodings from large T5

text_embeds = torch.randn(4, 256, 768).cuda()
images = torch.randn(4, 3, 256, 256).cuda()

# feed images into imagen, training each unet in the cascade

for i in (1, 2):
    loss = imagen(images, text_embeds = text_embeds, unet_number = i)
    loss.backward()

# do the above for many many many many steps
# now you can sample an image based on the text embeddings from the cascading ddpm

images = imagen.sample(texts = [
    'a whale breaching from afar',
    'young girl blowing out candles on her birthday cake',
    'fireworks with blue and green sparkles'
], cond_scale = 3.)

images.shape # (3, 3, 256, 256)

**Comment by xiankgx** [comment]
The code in the above is just to show you how to use the models. You need to do the "do the above for many many many many steps*. Also, the above images and text_embeds are *mock*.

**Comment by gyNancy** [comment]
> 

Thanks! When you mean the above images and text_embeds are mock, do you mean I should train it using my own dataset? I was initially just planned to see the image quality at first although the starting image is random. When you say many many... many steps, do you have any suggestions on the number of training steps?

**Comment by xiankgx** [comment]
Yes, you need to use your own images and train on your dataset. I do not have any idea for the desired number of steps but I would suggest at least 500k steps before giving up. I'm observing the training loss patterns of rapidly decreasing loss and then rather flat line in these diffusion models (GLIDE, Dalle2, this Imagen).

**Comment by gyNancy** [comment]
> Yes, you need to use your own images and train on your dataset. I do not have any idea for the desired number of steps but I would suggest at least 500k steps before giving up. I'm observing the training loss curve pattern of rapidly decreasing loss and then rather flat line in these diffusion models (GLIDE, Dalle2, this Imagen).

Thanks and this is really helpful! At least I got some sense on what is expected.

**Comment by 834799106** [comment]
> Yes, you need to use your own images and train on your dataset. I do not have any idea for the desired number of steps but I would suggest at least 500k steps before giving up. I'm observing the training loss curve pattern of rapidly decreasing loss and then rather flat line in these diffusion models (GLIDE, Dalle2, this Imagen).

Hi, may I ask if this is the loss curve you refer to? I have about 16,000 data, and 30,000 steps have been trained so far, but the generated details are still poor, and even the same results generated every 500 steps are quite different. I don't know whether to continue the training and what influence cond_scale has on the generated results.
![W B Chart 2022_8_19 18_34_55](https://user-images.githubusercontent.com/78247190/185600821-50eec075-2fca-4f6f-bc1a-f33b7ecb555d.png)

**Comment by xiankgx** [comment]
I just started yesterday, about the same I think. I'm just using about 10k images as testing. Batch size 16, img size 128, unet base dim 128. This is using the elucidated imagen. The loss is an order of magnitude lower with the normal imagen.

![Uploading image.png…]()

**Comment by deepglugs** [comment]
losses are not compatible across Imagen/Elucidated.  They use different calculations.
